src/common/snippets/src/pass_manager.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass_manager.hpp"
+
+namespace ov {
+namespace snippets {
+namespace pass {
+
+Manager::PassPosition::pass_list_type::const_iterator
+Manager::PassPosition::get_insert_position(const pass_list_type& pass_list) const {
+    if (m_pass_name.empty())
+        return m_after ? pass_list.cend() : pass_list.cbegin();
+    size_t pass_count = 0;
+    auto match = [this, &pass_count](const std::shared_ptr<PassBase>& p) {
+        auto name = p->get_name();
+        // Note that MatcherPass and ModelPass currently have different naming policies:
+        // - MatcherPass have names without namespaces, e.g. ConvertToSwishCPU
+        // - Similar ModelPass name includes namespaces: ov::snippets::pass::ConvertToSwishCPU
+        // So we have to remove everything before the last ':', and ':' itself
+        if (name.size() > m_pass_name.size()) {
```

Comment:
Should we verify the namespaces as well? For example, we might have passes with the same names but in different namespaces.

Reply:
It's a valid point, but unfortunately we can't.
First, let's have a look at the [ov::pass::PassBase::get_name()](https://github.com/openvinotoolkit/openvino/blob/04171416f4c1326d6434121ec2edb8f032a7202d/src/core/src/pass/pass.cpp#L32-L47). As you can see from the code, if m_name is empty it is filled through demangling and thus contains namespaces. This is the case for model passes, so their names always include namespaces, but MatcherPass names are formed differently (don't ask me why 🙃 )!
Let's have a look at [this MatcherPass](https://github.com/openvinotoolkit/openvino/blob/9fafcabb7c7b221821b0b77616d9967c8d13223f/src/common/snippets/src/pass/broadcast_to_movebroadcast.cpp#L16-L48) for example. `MATCHER_SCOPE(...)` (on L16) simply creates `matcher_name` variable and initializes it with its argument, and then the variable is then passed to the `Matcher` constructor (on L46). Then the interesting thing happen: MatcherPass takes the Matcher name and [assigns it to itself](https://github.com/openvinotoolkit/openvino/blob/af8e41bcea351817bb8d80f36e5a6894df60c80b/src/core/src/pass/graph_rewrite.cpp#L271) inside `register_matcher` (L47). So the `MatcherPass` names are (almost?) always equal to the corresponding class names and contain no namespaces.